packages/library-legacy-sham/src/__tests__/key-pair-test.ts

```diff
@@ -0,0 +1,81 @@
+import { utils } from '@noble/ed25519';
+
+import { Keypair } from '../key-pair';
+import { PublicKey } from '../public-key';
+
+const MOCK_PRIVATE_KEY_BYTES = [
+    151, 227, 37, 180, 104, 169, 5, 53, 191, 115, 132, 187, 223, 228, 25, 52, 7, 50, 86, 18, 151, 45, 105, 68, 31, 21,
+    128, 21, 32, 16, 222, 239,
+];
+const MOCK_PUBLIC_KEY_BYTES = [
+    117, 62, 75, 185, 26, 65, 209, 23, 95, 56, 97, 216, 197, 215, 208, 14, 138, 142, 59, 114, 43, 60, 190, 86, 21, 58,
+    46, 232, 77, 145, 46, 101,
+];
+
+describe('KeypairSham', () => {
+    it.each(['fromSecretKey', 'fromSeed'] as (keyof typeof Keypair)[])('throws when calling `%s`', method => {
```

Comment:
I expect people will also complain about these common methods not being implemented. Especially `fromSeed` as it allows you to create derived wallets on the fly and reduce the amount of approval popups your users see (to improve the UX).

Reply:
https://github.com/solana-labs/solana-web3.js/pull/1861#discussion_r1399770547